modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/PartitionAccess.java

```diff
@@ -197,4 +198,20 @@ public interface PartitionAccess {
      * @throws StorageRebalanceException If there are errors when trying to finish rebalancing.
      */
     CompletableFuture<Void> finishRebalance(long lastAppliedIndex, long lastAppliedTerm, RaftGroupConfiguration raftGroupConfig);
+
+    /**
+     * Returns the row ID for which the index needs to be built, {@code null} means that the index building has completed.
+     *
+     * @param indexId Index ID of interest.
+     * @throws StorageException If failed to get the row ID.
+     */
+    @Nullable RowId getNextRowIdToBuildIndex(int indexId);
```

Comment:
Is this method ever used?

Reply:
There `org.apache.ignite.internal.table.distributed.raft.snapshot.outgoing.SnapshotMetaUtils#collectNextRowIdToBuildIndexes`